DEVELOPING.md

```diff
@@ -70,21 +70,82 @@ For other commands, see
 
 Note at this stage, you will be unable to run Python commands directly. For example, ``pytest ./treeple`` will not work.
 
-However, after installing and building the project from source using meson, you can leverage editable installs to make testing code changes much faster. For more information on meson-python's progress supporting editable installs in a better fashion, see <https://meson-python.readthedocs.io/en/latest/how-to-guides/editable-installs.html>.
+However, after installing and building the project from source using meson, you can leverage editable installs to make testing code changes much faster.
 
-    pip install --no-build-isolation --editable .
+    spin install
 
-**Note: editable installs for treeple REQUIRE you to have built the project using meson already.** This will now link the meson build to your Python runtime. Now if you run
+This will now link the meson build to your Python runtime. Now if you run
 
     pytest ./treeple
 
 the unit-tests should run.
 
+Summary: Building locally with Meson (For developers)
+-----------------------------------------------------
+Make sure you have the necessary packages installed
+
+    # install build dependencies
+    pip install -r build_requirements.txt
+
+    # you may need these optional dependencies to build scikit-learn locally
+    conda install -c conda-forge joblib threadpoolctl pytest compilers llvm-openmp
+
+We use the ``spin`` CLI to abstract away build details:
+
+    # run the build using Meson/Ninja
+    ./spin build
+
+    # you can run the following command to see what other options there are
+    ./spin --help
+    ./spin build --help
+
+    # For example, you might want to start from a clean build
+    ./spin build --clean
+
+    # or build in parallel for faster builds
+    ./spin build -j 2
+
+    # you will need to double check the build-install has the proper path
+    # this might be different from machine to machine
+    export PYTHONPATH=${PWD}/build-install/usr/lib/python3.9/site-packages
```

Comment:
might wanna replace `python3.9` with some placeholder value like `YOUR_PYTHON_VERSION`, no?  wouldn't it just depend on what Python your environment is?

Reply:
Good point. Done in 088f9cfb